### PR TITLE
Data rewrite

### PR DESCRIPTION
**Pull request by tmonfre**
# Description

Updates database fetching for data rewrite in automation repo (https://github.com/dali-lab/pine-beetle-automation/pull/133).

Only have routers for summarized, unsummarized, and admin user data. Summarized routers support the ability to run aggregation pipelines (MongoDB's version of group by and calculated columns). Added routes for aggregating by year and computing averages/sums/etc. Also support aggregating by state, county, and ranger district. Also added routes for generating list of years, states, counties, and ranger districts represented in data.

All routes support optional `startYear`, `endYear`, `state`, `county`, and `rangerDistrict` query params for filtering. State, county, and ranger districts can include multiple locations, comma separated.

## Type of Change

- [x] New feature
- [ ] Bug fix
- [x] Breaking change
- [ ] Documentation update

### CODE REVIEW

**Review by AlejandroALopez — APPROVED** [review]
Thanks for the huge help!

File: src/utils/aggregate.js [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        ...projectComputedFields(),
+      },
+    },
+  ];
+}
+
+/**
+ * @description generates a mongoDB aggregation pipeline to list all available years in data
+ * @param {String} location either 'county' or 'rangerDistrict'
+ * @param {Object} [filters={}] optional filters for isHistorical and isPrediction
```
Comment (jeffzyliu) [inline src/utils/aggregate.js]: Did you intend to copy the jsdocs for each filter from above to here?

File: src/utils/aggregate.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    },
+    { $group: { _id: '$year' } },
+    { $sort: { _id: 1 } },
+    { $project: { _id: 0, year: '$_id' } },
+  ];
+}
+
+/**
+ * @description generates a mongoDB aggregation pipeline to list all available states in data
+ * @param {Object} [filters={}] optional filters for isHistorical and isPrediction
```
Comment (jeffzyliu) [inline src/utils/aggregate.js]: Same here

File: src/utils/aggregate.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    { $group: { _id: '$state' } },
+    { $sort: { _id: 1 } },
+    { $project: { _id: 0, state: '$_id' } },
+  ];
+}
+
+/**
+ * @description generates a mongoDB aggregation pipeline to list all available counties or ranger districts in data
+ * @param {String} location either 'county' or 'rangerDistrict'
+ * @param {Object} [filters={}] optional filters for isHistorical and isPrediction
```
Comment (jeffzyliu) [inline src/utils/aggregate.js]: Here

File: src/utils/aggregate.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+/**
+   * @description generates a mongoDB aggregation pipeline by year
+   * @param {String} location either 'county' or 'rangerDistrict'
+   * @param {Number} [startYear] optional start year to filter on
+   * @param {Number} [endYear] optional end year to filter on
+   * @param {String} [state] optional state to filter on
+   * @param {String} [loc] optional sublocation to filter on (name of county or ranger district)
+   * @returns
+   */
+export function generateStatePipeline(location, startYear, endYear, state, loc) {
```
Comment (jeffzyliu) [inline src/utils/aggregate.js]: Do you want to make the interface the same using filters object in this function?

File: src/utils/aggregate.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+/**
+   * @description generates a mongoDB aggregation pipeline by year
+   * @param {String} location either 'county' or 'rangerDistrict'
+   * @param {Number} [startYear] optional start year to filter on
+   * @param {Number} [endYear] optional end year to filter on
+   * @param {String} [state] optional state to filter on
+   * @param {String} [loc] optional sublocation to filter on (name of county or ranger district)
+   * @returns
+   */
+export function generateLocationPipeline(location, startYear, endYear, state, loc) {
```
Comment (jeffzyliu) [inline src/utils/aggregate.js]: Same comment about filter object interface

**Review by jeffzyliu — APPROVED** [review]

**Pull request merged**